ot/mpz-ot/src/kos/sender.rs

```diff
@@ -0,0 +1,312 @@
+    >(
+        &mut self,
+        sink: &mut Si,
+        stream: &mut St,
+    ) -> Result<(), SenderError> {
+        // If the sender is committed, we sample delta using a cointoss.
+        let delta = if self.state.as_initialized()?.config().sender_commit() {
+            let (cointoss_sender, commitment) =
+                cointoss::Sender::new(vec![thread_rng().gen()]).send();
+
+            sink.send(Message::CointossCommit(commitment)).await?;
+            let payload = stream
+                .expect_next()
+                .await?
+                .into_cointoss_receiver_payload()?;
+
+            let (seeds, payload) = cointoss_sender.finalize(payload)?;
+
+            // Store the payload to reveal to the receiver later.
+            self.cointoss_payload = Some(payload);
+
+            seeds[0]
+        } else {
+            Block::random(&mut thread_rng())
+        };
+
+        self._setup_with_delta(sink, stream, delta).await
+    }
+
+    /// Performs the base OT setup with the provided delta.
+    ///
+    /// # Arguments
+    ///
+    /// * `sink` - The sink to send messages to the base OT sender
+    /// * `stream` - The stream to receive messages from the base OT sender
+    /// * `delta` - The delta value to use for the base OT setup.
+    pub async fn setup_with_delta<
+        Si: IoSink<Message<BaseOT::Msg>> + Send + Unpin,
+        St: IoStream<Message<BaseOT::Msg>> + Send + Unpin,
+    >(
+        &mut self,
+        sink: &mut Si,
+        stream: &mut St,
+        delta: Block,
+    ) -> Result<(), SenderError> {
+        if self.state.as_initialized()?.config().sender_commit() {
+            return Err(SenderError::ConfigError(
+                "committed sender can not choose delta".to_string(),
+            ));
+        }
+
+        self._setup_with_delta(sink, stream, delta).await
+    }
+
+    async fn _setup_with_delta<
+        Si: IoSink<Message<BaseOT::Msg>> + Send + Unpin,
+        St: IoStream<Message<BaseOT::Msg>> + Send + Unpin,
+    >(
+        &mut self,
+        sink: &mut Si,
+        stream: &mut St,
+        delta: Block,
+    ) -> Result<(), SenderError> {
+        let ext_sender = self.state.replace(State::Error).into_initialized()?;
+
+        let choices = delta.into_lsb0_vec();
+        let seeds = self
+            .base
+            .receive(
+                &mut into_base_sink(sink),
+                &mut into_base_stream(stream),
+                &choices,
+            )
+            .await?;
+
+        let seeds: [Block; CSP] = seeds.try_into().expect("seeds should be CSP length");
+
+        let ext_sender = ext_sender.setup(delta, seeds);
+
+        self.state = State::Extension(ext_sender);
+
+        Ok(())
+    }
+
+    /// Performs OT extension.
+    ///
+    /// # Arguments
+    ///
+    /// * `channel` - The channel to communicate with the receiver.
+    /// * `count` - The number of OTs to extend.
+    pub async fn extend<
```

Comment:
Currently re-extension is not supported, right? because on each call to extend it performs a cointoss here. Maybe it's worth opening a gh issue for now to add re-extension support later.